src/components/ebay-combobox/index.js

```diff
@@ -0,0 +1,256 @@
+const eventUtils = require('../../common/event-utils');
+const processHtmlAttributes = require('../../common/html-attributes');
+const observer = require('../../common/property-observer');
+const template = require('./template.marko');
+
+const comboboxOptionsClass = 'combobox__options';
+const comboboxExpanderClass = 'combobox__control';
+const comboboxHostSelector = `.${comboboxExpanderClass} > input`;
+const comboboxBtnClass = 'combobox__control';
+const comboboxOptionSelector = '.combobox__option[role=option]';
+
+function getInitialState(input) {
+    const options = (input.options || []).map(option => {
+        const selected = option.selected;
+
+        return {
+            class: option.class,
+            value: option.value,
+            label: option.label,
+            selected: Boolean(selected),
+            htmlAttributes: processHtmlAttributes(option),
+            renderBody: option.renderBody
+        };
+    });
+
+    const selectedOption = options.filter(option => option.selected)[0] || options[0];
+
+    if (options.length > 0 && selectedOption.value === options[0].value) {
+        options[0].selected = true;
+    }
+
+    return {
+        name: input.name,
+        class: input.class,
+        options,
+        selected: selectedOption,
+        grow: input.grow,
+        disabled: input.disabled,
+        borderless: Boolean(input.borderless),
+        htmlAttributes: processHtmlAttributes(input)
+    };
+}
+
+function getTemplateData(state) {
+    const comboboxClass = ['combobox', state.class];
+    const btnClass = [comboboxBtnClass];
+    const optionsClass = [comboboxOptionsClass];
+
+    if (state.borderless) {
+        btnClass.push('combobox__control--borderless');
+    }
+
+    return {
+        class: comboboxClass,
+        btnClass,
+        optionsClass,
+        name: state.name,
+        selectedOption: state.selected,
+        options: state.options,
+        grow: state.grow,
+        disabled: state.disabled,
+        htmlAttributes: state.htmlAttributes
+    };
+}
+
+function init() {
+    const optionEls = this.el.querySelectorAll(comboboxOptionSelector);
+
+    if (this.state.options && this.state.options.length > 0) {
+        this.createExpander();
+
+        observer.observeRoot(this, ['selected'], (index) => {
+            this.processAfterStateChange(optionEls[index]);
+        });
+
+        const selectedObserverCallback = (optionEl) => {
+            this.processAfterStateChange(optionEl);
+        };
+
+        this.optionEls = optionEls.forEach((optionEl, i) => {
+            observer.observeInner(this, optionEl, 'selected', `options[${i}]`, 'options', selectedObserverCallback);
+        });
+
+        scrollKeyPreventer.add(this.el.querySelector(comboboxHostSelector));
+        scrollKeyPreventer.add(this.el.querySelector(`.${comboboxOptionsClass}`));
+    }
+}
+
+function onUpdate() {
+    if (!this.expander && this.state.options && this.state.options.length > 0) {
```

Comment:
This will still break if someone is toggling `disabled` on the component. We need to (onUpdate) check to ensure the the expander is enabled or disabled depending on the current `disabled` state.

Reply:
Yeah, I pushed this but am still working on it.